src/app/container/deregister-modal/deregister-modal.component.ts

```diff
@@ -27,28 +28,45 @@ import { RegisterToolService } from './../register-tool/register-tool.service';
   styleUrls: ['./deregister-modal.component.css']
 })
 export class ModalComponent implements OnInit {
+  @Input() confirmationDialogData: ConfirmationDialogData;
+  @Input() toolTipMessage: string;
   @Input() refreshMessage: boolean;
+  @Input() option: string;
+  @Input() organizationID: number;
+  @Input() organizationStatus: string;
+  @Input() organizationName: string;
 
-  constructor(private registerToolService: RegisterToolService, private confirmationDialogService: ConfirmationDialogService) {}
+  constructor(
+    private registerToolService: RegisterToolService,
+    private confirmationDialogService: ConfirmationDialogService,
+    private requestsService: RequestsService
+  ) {}
 
   ngOnInit() {}
-  deregister() {
+
+  deregisterTool() {
     this.registerToolService.deregisterTool();
   }
 
+  deregisterOrganization(id: number) {
+    this.requestsService.deleteOrganization(id);
+  }
+
   openDeleteDialog() {
-    const confirmationDialogData: ConfirmationDialogData = {
-      title: 'Are you sure you wish to delete this tool?',
-      message: `All information associated with this tool will be deleted.`,
-      cancelButtonText: 'Cancel',
-      confirmationButtonText: 'Delete'
-    };
     this.confirmationDialogService
-      .openDialog(confirmationDialogData, bootstrap4mediumModalSize)
+      .openDialog(this.confirmationDialogData, bootstrap4mediumModalSize)
       .pipe(first())
       .subscribe(result => {
+        console.log(this.organizationID);
```

Comment:
Remove when done

Reply:
No longer using deregister-modal. Created a new method to display a new delete dialog for requests.component.